### ISSUE TITLE
Sometimes I can't navigate easily from my current position ("Add a starting point to plan a route")

### ISSUE THREAD

**Issue by kuba-orlik** [organicmaps/organicmaps#2883]
Usually when I pick a POI on the map and tap the "Route to" button, the app immediately starts planning the route from my current position. But sometimes it does something else and instead of planning the route from my current position, it shows a "Add a starting point to plan a route" bar on the bottom of the screen:

![photo_2022-06-30_21-17-08](https://user-images.githubusercontent.com/2697916/176760351-ba1b2316-7375-4704-b1f4-62c643f556c9.jpg)

I can't easily pick the "navigate from my current position" option, so I have to manually scroll to my current location, tap on it, and then pick the "route from"

**Comment by biodranik** [comment]
This happens because your current position is unknown at that moment. Please detect it first by pressing a crosshair button, and then build a route.

**Comment by kuba-orlik** [comment]
I always thought that the app detects my location constantly and the crosshair button only centers the map on the users location, is that not the case?

**Comment by biodranik** [comment]
That has been refactored. If location detection stopped for whatever reason, it is not automatically started after the app restart, to avoid wasting the battery and annoying users when it's not available (at home, in airplane mode, etc.)

**Comment by bitboy85** [comment]
Is it detectable if it is a device with a battery or a constant powered device like a car head unit?

**Comment by biodranik** [comment]
> Is it detectable if it is a device with a battery or a constant powered device like a car head unit?

Yes

**Comment by kuba-orlik** [comment]
I think that in such scenario it could be cool to have a "start from my current location" button. Otherwise I have to cancel the route building, detect my location manually, then perform the search again and press the "route to" button. That's a lot of steps! 

And this situation often happens when I really don't expect it - I'm on the open air, good GPS signal, didn't restart my phone or the app recently, and suddenly the "route to" button doesn't behave as usual

**Comment by biodranik** [comment]
@kuba-orlik no, it will complicate the UX. A simpler solution is to display another icon instead of the current crosshair. It does not distinguish between different states:
1. No position and it is not detected at the moment
2. There is a position, but it's not visible on the current screen
3. No position yet, but its detection is in progress.

**Comment by bitboy85** [comment]
> > Is it detectable if it is a device with a battery or a constant powered device like a car head unit?
> 
> Yes

Would be glad to have location detection constantly running on non-battery devices. It is somewhat unexpected to manually start detection if the first fails.
As an Alternative, have a greater time out. Like turning it off if it doesn`t find the position after 3 minutes.

**Comment by biodranik** [comment]
That's a good idea. Timeout can be longer if there is battery power. @rtsisyk @vng

**Comment by euf** [comment]
Maybe it would be better for UX to try this approach: 
1. User taps a "Route to" button 
2. If location detection was off, restart it 
3. Wait for a reasonable amount of time (I believe a couple of seconds should suffice for the current battery saving scenario)
4. If current location became available, build a route from it (as @kuba-orlik expects)
5. Otherwise, proceed with “Add a starting point” dialogue (in case location services are off, user is on an airplane, etc) 

It could be further improved: if the app somehow knows from the system that location is not available, no need to wait at all.

**Comment by biodranik** [comment]
(3) A cold start of GPS may take up to several minutes. Only if the location was already detected by some other app, it will work fast. In other cases, if the Internet is available it _may_ take a couple of seconds.
